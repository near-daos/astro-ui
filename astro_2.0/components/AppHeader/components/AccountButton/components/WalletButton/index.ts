export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
