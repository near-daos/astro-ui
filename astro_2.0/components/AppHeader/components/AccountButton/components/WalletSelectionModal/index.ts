export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
