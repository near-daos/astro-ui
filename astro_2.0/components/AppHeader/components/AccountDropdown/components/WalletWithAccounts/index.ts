export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
