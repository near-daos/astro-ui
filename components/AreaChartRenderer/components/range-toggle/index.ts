export * as RangeToggle from './RangeToggle';
