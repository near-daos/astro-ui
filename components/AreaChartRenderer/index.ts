import {
  AreaChartRenderer,
  AreaChartProps,
} from 'components/AreaChartRenderer/AreaChartRenderer';

export default AreaChartRenderer;
export type { AreaChartProps };
