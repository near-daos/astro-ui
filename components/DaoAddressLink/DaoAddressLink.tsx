import React, { MouseEvent } from 'react';
import { configService } from 'services/ConfigService';

interface DaoAddressLinkProps {
  daoAddress: string;
}

export const DaoAddressLink: React.VFC<DaoAddressLinkProps> = ({
  daoAddress,
}) => {
  const { nearConfig } = configService.get();

  const daoAddressLink = `${nearConfig?.explorerUrl}/accounts/${daoAddress}`;

  function stopPropagation(e: MouseEvent) {
    e.stopPropagation();
  }

  return (
    <>
      {daoAddress && (
        <a
          href={daoAddressLink}
          onClick={stopPropagation}
          target="_blank"
          rel="noreferrer"
        >
          {daoAddress}
        </a>
      )}
    </>
  );
};
