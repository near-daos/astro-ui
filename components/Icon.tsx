import React from 'react';
import getConfig from 'next/config';

import * as icons from 'assets/icons';

export type IconName = keyof typeof icons;

export type IconProps = React.SVGProps<SVGSVGElement> & {
  name: IconName;
  title?: string;
};

export const Icon: React.VFC<IconProps> = ({ name, title, ...svgProps }) => {
  const { viewBox, id, url } = icons[name];

  const config = getConfig();

  return (
    <svg viewBox={viewBox} {...svgProps}>
      {title && <title> {title}</title>}
      <use
        href={
          config?.publicRuntimeConfig?.shouldUseExternalAssetUrl
            ? `${config?.publicRuntimeConfig?.assetUrl}#${id}`
            : `/_next/${url}`
        }
      />
    </svg>
  );
};
