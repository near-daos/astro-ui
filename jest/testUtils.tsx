import {
  RenderResult,
  render as reactTestingLibRender,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';

function createMockRouter(router?: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    query: {},
    asPath: '/',
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isLocaleDomain: false,
    isReady: true,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}

export function render(
  component: ReactNode,
  router?: Partial<NextRouter>
): RenderResult {
  return reactTestingLibRender(
    <RouterContext.Provider value={createMockRouter(router)}>
      {component}
    </RouterContext.Provider>
  );
}
