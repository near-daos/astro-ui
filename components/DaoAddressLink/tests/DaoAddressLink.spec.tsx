import { render } from 'jest/testUtils';

import { DaoAddressLink } from 'components/DaoAddressLink';
import { fireEvent } from '@testing-library/dom';

describe('DaoAddressLink', () => {
  it('Should render nothing if no dao address', () => {
    const { container } = render(<DaoAddressLink daoAddress="" />);

    expect(container).toMatchSnapshot();
  });

  it('Should render component', () => {
    const { container } = render(<DaoAddressLink daoAddress="HelloWorld!" />);

    expect(container).toMatchSnapshot();
  });

  it('Should call stopPropagation on click', () => {
    const daoAddress = 'HelloWorld!';

    jest
      .spyOn(Event.prototype, 'stopPropagation')
      .mockImplementationOnce(jest.fn());

    const { getByText } = render(<DaoAddressLink daoAddress="HelloWorld!" />);

    fireEvent.click(getByText(daoAddress));

    expect(Event.prototype.stopPropagation).toBeCalled();
  });
});
