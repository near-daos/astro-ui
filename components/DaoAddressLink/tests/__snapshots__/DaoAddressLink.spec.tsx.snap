// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaoAddressLink Should render component 1`] = `
<div>
  <a
    href="https://explorer.testnet.near.org/accounts/HelloWorld!"
    rel="noreferrer"
    target="_blank"
  >
    HelloWorld!
  </a>
</div>
`;

exports[`DaoAddressLink Should render nothing if no dao address 1`] = `
<div>
  
</div>
`;
